runtime/engine/src/test/java/io/aklivity/zilla/runtime/engine/test/EngineRule.java

```diff
@@ -171,7 +172,67 @@ public EngineExtContext context()
         return engine.context();
     }
 
+    public long[][] counterIds()
+    {
+        return engine.counterIds();
+    }
+
+    public LongSupplier counter(
+        long bindingId,
+        long metricId)
+    {
+        return engine.counter(bindingId, metricId);
+    }
+
     public LongConsumer counterWriter(
+        long bindingId,
+        long metricId,
+        int core)
+    {
+        return engine.counterWriter(bindingId, metricId, core);
+    }
+
+    public long[][] gaugeIds()
+    {
+        return engine.gaugeIds();
+    }
+
+    public LongSupplier gauge(
+        long bindingId,
+        long metricId)
+    {
+        return engine.gauge(bindingId, metricId);
+    }
+
+    public LongConsumer gaugeWriter(
+        long bindingId,
+        long metricId,
+        int core)
+    {
+        return engine.gaugeWriter(bindingId, metricId, core);
+    }
+
+    public long[][] histogramIds()
+    {
+        return engine.histogramIds();
+    }
+
+    public LongSupplier[] histogram(
+        long bindingId,
+        long metricId)
+    {
+        return engine.histogram(bindingId, metricId);
+    }
+
+    public LongConsumer histogramWriter(
+        long bindingId,
+        long metricId,
+        int core)
+    {
+        return engine.histogramWriter(bindingId, metricId, core);
+    }
+
+    public LongConsumer contextCounterWriter(
```

Comment:
This doesn't need to be renamed, can stay as `counterWriter` for now, agree?

Reply:
Let's keep things simple rather than exposing internal implementation detail in the method name and call it `counterWriter`.

Reply:
done